### ISSUE TITLE
Not all Language Servers detected

### ISSUE THREAD

**Issue by le4ker** [codicocodes/dotfyle#89]
Thank you for Dotfyle, it's great!

One potential bug is that it can't detected all language servers, at least in my config.
For example [here](https://dotfyle.com/le4ker/nvim-config) it detects the following language servers:

- tflint
- html
- clangd
- marksman
- gopls
- graphql
- pyright

And it's missing the following ones:

- bash-language-server
- marksman
- terraform-ls
- yamlls

**Comment by le4ker** [comment]
Opened a [PR](https://github.com/codicocodes/dotfyle/pull/90) in case the `html` and `graphql` must be removed.

**Comment by le4ker** [comment]
Seems like it might be parsing the [overrides.lua](https://github.com/le4ker/nvim-config/blob/v2.0/lua/custom/configs/overrides.lua) file but not the [lspconfig.lua](https://github.com/le4ker/nvim-config/blob/v2.0/lua/custom/configs/lspconfig.lua)? Because all the missing LSPs are in my lspconfig, but they have a different name in the Mason config 😄

**Comment by codicocodes** [comment]
html and graphql are taken from here: https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md not from mason

**Comment by le4ker** [comment]
@codicocodes thank you, I closed my PR. Why does dotfyle detect those LSPs though in my case? Also why aren't the rest LSPs detected, even though they are defined in the known language servers?

**Comment by codicocodes** [comment]
Detection is looking for the string 'html' or 'graphql' in your config, this works for most language servers pretty well, but for html could easily have false positives. Here is your HTML setup btw, so it is not a false positive I believe: https://github.com/le4ker/nvim-config/blob/v2.0/lua/custom/configs/lspconfig.lua#L40-L43 The graphql one could be a false positive.

Not sure about why some are not detected though, it looks like that could very well be a bug.

**Comment by le4ker** [comment]
Yeah I added the html LSP a couple of days ago, but if was a false positive like graphql before that.

I noticed as well that in the code it just greps for the strings, so it's strange indeed that it can't detect some LSPs 🤔

**Comment by iainsimmons** [comment]
Part of the issue here is that there's two different names used for some language servers, and Mason's `ensure_installed` takes a different format than the ones in the [known language servers](https://github.com/codicocodes/dotfyle/blob/main/src/lib/server/nvim-sync/config/LanguageServerFinder.ts).

See https://github.com/iainsimmons/nvim-config/blob/main/lua/plugins/lsp.lua#L182-L202

Also, you might get some false positives if the LSP name is matched in a code comment, such as for those using the [linting setup from kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim/blob/master/lua/kickstart/plugins/lint.lua).

Or if the filename matches the name of the language server (as was the case for astro, html and svelte in my config) and is used elsewhere (e.g. for my [emmet_ls config](https://github.com/iainsimmons/nvim-config/blob/main/lua/plugins/lsp.lua#L145-L162)).

My first pass with Dotfyle generated the following:

+ astro
+ glint
+ html
+ lemminx
+ marksman
+ svelte
+ tflint

Some of which are the ones that don't use a different format for the language server name.

Here are some that are different (Mason / LSP Config):
+ "astro-language-server" / "astro"
+ "emmet-ls" / "emmet_ls"
+ "eslint-lsp" / "eslint"
+ "html-lsp" / "html"
+ "htmx-lsp" / "htmx"
+ "json-lsp" / "jsonls"
+ "lua-language-server" / "lua_ls"
+ "svelte-language-server" / "svelte"
+ "tailwindcss-language-server" / "tailwindcss"
+ "typescript-language-server" / "tsserver"
+ "yaml-language-server" / "yamlls"

So you almost need to match all the Mason strings too.

**Comment by le4ker** [comment]
@cod### eid you make any updates related to this issue? It seems like it’s detecting way more than before 😁
